app/controllers/questions_controller.rb

```diff
@@ -1,27 +1,89 @@
 class QuestionsController < ApplicationController
   def show
-    @user = current_user
-    user_id = @user.id
 
+    @user = current_user
     @question = LearningObject.find(params[:id])
-    rel = @question.seen_by_user(user_id)
+    rel = @question.seen_by_user(current_user.id)
     gon.userVisitedLoRelationId = rel.id
-    @next_question = @question.next(params[:week_number])
-    @previous_question = @question.previous(params[:week_number])
+    @next_question = @question.next(params[:room_number],@question.id)
+    @previous_question = @question.previous(params[:room_number])
+
+    @room = Room.find(params[:room_number])
 
     @answers = @question.answers
-    @relations = UserToLoRelation.where(learning_object_id: params[:id], user_id: user_id).group('type').count
+    @relations = UserToLoRelation.where(learning_object_id: params[:id], user_id: current_user.id).group('type').count
 
     if @user.show_solutions
-      UserViewedSolutionLoRelation.create(user_id: user_id, learning_object_id: params[:id], setup_id: 1, )
+      UserViewedSolutionLoRelation.create(user_id: current_user.id, learning_object_id: params[:id], setup_id: 1, )
       solution = @question.get_solution(current_user.id)
       gon.show_solutions = TRUE
       gon.solution = solution
     end
 
+    # Poradie zobrazenia danej otazky
+    if @user.show_solutions
+      @number_of_question = @room.learning_objects.where("learning_object_id < ?",@question.id).count + 1
+    else
+      @number_of_question = @room.question_count - @room.question_count_not_visited
+      if @room.not_answered(@question.id)
+        @number_of_question +=1
+      end
+    end
+
     @feedbacks = @question.feedbacks.visible.includes(:user)
   end
 
+  # Metoda sluzi na vypocet skore za odpoved na otazku
+  def eval_question
+    setup = Setup.take
+    room = Room.find(params[:room_number])
+
+    lo_class = Object.const_get params[:type]
+    lo = lo_class.find(params[:id])
+
+    results_used = UserToLoRelation.get_results_room(current_user.id,params[:week_number],params[:room_number], room.number_of_try,lo.id).try(:first)
+    used = FALSE
+    unless results_used.nil?
+      used = results_used["solved"] != 0 || results_used["failed"] != 0 || results_used["donotnow"] != 0
+    end
+
+    room = Room.find(params[:room_number])
+
+    if params[:commit] == 'send_answer'
+      result = lo.right_answer? params[:answer], @solution
+    end
+
+    # Nacitanie parametrov  obtiaznosti a dolezitosti danej otazky
+    dif_result = lo.get_difficulty(setup)
+    imp_value = lo.get_importance
+
+    score = 0.0
+
+    if lo.type == "EvaluatorQuestion"
```

Comment:
pri tomto mozes vyuzit polymorfizmus (ano, je to mozno prekvapive, ale aj Ruby ho ma :)):
```
if lo.is_a?(EvaluatorQuestion)
```